src/specialOffer/SpecialOffer.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react'
+
+import { Button, ButtonStatus } from '../button'
+import { ButtonGroup } from '../buttonGroup'
+import { Disclaimer } from '../disclaimer'
+import { Paragraph } from '../paragraph'
+import { SubHeader } from '../subHeader'
+import { StyledSpecialOffer, StyledSpecialOfferPicture } from './SpecialOffer.style'
+
+type Picture = Readonly<{
+  url: string
+  alt: string
+}>
+
+type Link = Readonly<{
+  label: string
+  url: string
+}>
+
+type Button = Readonly<{
+  label: string
+  onClick: () => void
+}>
+
+type CTALinks = [Link] | [Link, Link]
+type CTAButtons = [Button] | [Button, Button]
+
+type SpecialOfferBaseProps = Readonly<{
+  picture: Picture
+  title: string
+  description: string
+  informationLink?: Link
+}>
+
+export type SpecialOfferLinksProps = Readonly<
+  SpecialOfferBaseProps & {
+    ctaLinks: CTALinks
+  }
+>
+export type SpecialOfferButtonsProps = Readonly<
+  SpecialOfferBaseProps & {
+    ctaButtons: CTAButtons
+  }
+>
+
+function SpecialOffer(props: SpecialOfferLinksProps): JSX.Element
+function SpecialOffer(props: SpecialOfferButtonsProps): JSX.Element
+function SpecialOffer({
+  picture,
+  title,
+  description,
+  informationLink,
+  ctaLinks,
+  ctaButtons,
+}: SpecialOfferBaseProps &
+  Readonly<{
+    ctaLinks?: CTALinks
+    ctaButtons?: CTAButtons
+  }>) {
+  return (
+    <StyledSpecialOffer>
+      <StyledSpecialOfferPicture src={picture.url} alt={picture.alt} />
+      <SubHeader>{title}</SubHeader>
```

Comment:
Make sure title is available to avoid rendering a empty element

Reply:
Typescript is here to secure that. Adding a special check (`if` of `&&`) would make the code more complex without adding any more security, right?

Reply:
that means title is **_required_** than

Reply:
Yes it is, as typed here: https://github.com/blablacar/ui-library/blob/59e96c6cea8051973c77aedc5af820ef6ee3e41d/src/specialOffer/SpecialOffer.tsx#L10-L44

Everything is required appart from `informationLink`